content/en/blog/_posts/2021-01-28-PSP-deprecation.md

```diff
@@ -0,0 +1,50 @@
+---
+layout: blog
+title: 'What’s Up with Pod Security Policies?'
+date: 2021-02-04
+slug: psp-update-2021
+---
+
+**Authors:** Contributor Comms Team
+
+_The following is a brief overview of upcoming changes to Kubernetes. Bookmark and [keep an eye on our release site](https://www.kubernetes.dev/resources/release/) to see the latest news and other important updates._
+Pod Security Policies (PSP), starting with the Kubernetes 1.21, will begin the [process of deprecation](/docs/reference/using-api/deprecation-policy/) with the intention to fully remove it in a future release. This prompts the question: what will replace Pod Security Policies in the future? In short, [Admission Controllers](/docs/reference/access-authn-authz/admission-controllers/). For more details, keep reading.
+
+
+## What do Pod Security Policies do now?
+
+> A Pod Security Policy is a cluster-level resource that controls security sensitive aspects of the pod specification. The [PodSecurityPolicy](/docs/reference/kubernetes-api/policies-resources/pod-security-policy-v1beta1/) object defines a set of conditions that a pod must run with in order to be accepted into the system, as well as defaults for the related fields. They allow an administrator to control a number of security-related functions.
+
+The [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/)
+documentation explains this in more detail.
+
+
+## Why are Pod Security Policies being deprecated? 
+
+To watch an explanation of PSP removal, you can check out this session by SIG-Auth at KubeCon NA 2019:
+
+{{< youtube "SFtHRmPuhEw?start=953" youtube-quote-sm >}}
+
+In short, PSPs are part of the kubelet and API server. The tight coupling to these parts of Kubernetes makes PSPs cumbersome both for developers and for operators. Using PSPs is often unintuitive, and setting them up too often leads to unintended results. The initial design of PSPs did not foresee many of the use cases they’ve since tried to solve.
```

Comment:
I don't actually think there's a problem with the tight coupling to the API server. Both of the replacement designs we're considering are still built-in. [This document](https://docs.google.com/document/d/1VKqjUlpU888OYtIrBwidL43FOLhbmOD5tesYwmjzO4E/edit#heading=h.3qhd5uuytbqs) has a summary of what I see as the top issues with PSP. How about this as an alternative?
```suggestion
In short, there are 3 main challenges with PSPs today:
1. A flawed policy authorization model - Policies can be bound either to the pod's service account, or the requester. This dual binding approach weakens security, and often leads to confusion and breakage.
2. Rollout challenges - PSP is enabled or disabled for the entire cluster, lacks a non-enforcing or dryrun mode, and fails closed. This combination makes it very difficult to enable PSP by default, or rollout to an existing cluster.
3. Inconsistent API - PSP has grown organically over many releases but doesn't have a clear scope of what should be included. Backwards compatibility issues has lead to inconsistencies in the default behavior of several fields, and mutations can have unexpected side effects.
```